packages/suite/src/components/suite/DeviceDisplay.tsx

```diff
@@ -0,0 +1,221 @@
+    position: relative;
+    bottom: ${({ isPixelType }) => (isPixelType ? '13' : '20')}px;
+    right: ${({ isPixelType }) => (isPixelType ? '-95' : '35')}px;
+`;
+
+const StyledContinuesIcon = styled(Icon)<{ isPixelType: boolean }>`
+    position: relative;
+    top: ${({ isPixelType }) => (isPixelType ? '10' : '25')}px;
+    right: ${({ isPixelType }) => (isPixelType ? '82' : '97')}px;
+`;
+
+export interface DeviceDisplayProps {
+    address: string;
+    network?: string;
+}
+
+export const DeviceDisplay = ({ address, network }: DeviceDisplayProps) => {
+    const selectedDeviceInternalModel = useSelector(selectDeviceInternalModel);
+    const unavailableCapabilities = useSelector(selectDeviceUnavailableCapabilities);
+    const addressDisplayType = useSelector(selectAddressDisplayType);
+
+    const isCardano = network === 'cardano';
+    const areChunksUsed =
+        addressDisplayType === AddressDisplayOptions.CHUNKED &&
+        !unavailableCapabilities?.chunkify &&
+        !isCardano;
+    const isPixelType = selectedDeviceInternalModel !== DeviceModelInternal.T2T1;
+    const iconSize = isPixelType ? 10 : 20;
+    const iconColor = isPixelType ? '#ffffff' : '#959596';
+    const iconNextName = isPixelType ? 'ADDRESS_PIXEL_NEXT' : 'ADDRESS_NEXT';
+    const iconContinuesName = isPixelType ? 'ADDRESS_PIXEL_CONTINUES' : 'ADDRESS_CONTINUES';
+
+    const showChunksInRows = (chunks: ReactNode[][] | undefined, isNextAddress?: boolean) =>
+        chunks?.map((row, rowIndex) => (
+            <Row
+                // eslint-disable-next-line react/no-array-index-key
+                key={rowIndex}
+                isAlignedRight={rowIndex === 0 && isNextAddress}
+            >
+                {row.map((chunk, chunkIndex) => (
+                    // eslint-disable-next-line react/no-array-index-key
+                    <Text isPixelType={isPixelType} key={chunkIndex}>
+                        {chunk}
+                    </Text>
+                ))}
+            </Row>
+        ));
+
+    const renderChunks = (address: string) => {
+        const chunks = address.match(/.{1,4}/g);
+
+        if (isCardano) {
+            // Add pagination icons
+            // @ts-expect-error
+            chunks?.splice(15, 0, <Icon size={iconSize} icon={iconNextName} color={iconColor} />);
+            chunks?.splice(
+                16,
+                0,
+                // @ts-expect-error
+                <Icon size={iconSize} icon={iconContinuesName} color={iconColor} />,
+            );
+        }
+
+        const groupedChunks = chunks?.reduce((result, chunk, index) => {
+            const rowIndex = Math.floor(index / 4);
+
+            if (!result[rowIndex]) {
+                result[rowIndex] = [];
+            }
+
+            result[rowIndex].push(chunk);
+
+            return result;
+        }, [] as ReactNode[][]);
+
+        const handleOnCopy = (event: ClipboardEvent) => {
```

Comment:
This is applied to every usage of Chunks component, right? So it would make more sense to me to do it as part of the Chunks component. But it's a nitpick. Another thing that is more a matter of taste is that `renderChunks` and `renderOriginal` can be defined as component and not plain function, but it's up to you.

Reply:
I would probably keep it as it is now